### PR TITLE
refactor: simplify extractor

### PR DESCRIPTION
**Pull request by cmdoret**
This PR simplifies the extractor interface by adding a unique `Repository` model (see #71).
Until now, all repository metadata was stored directly in extractors, and each extractor had to define its own schema for RDF serialization. The aim of this PR is to decouple the extractor from the repository. It does the following:

* `Extractor`s only have the (public) methods `list_files()` and `extract()`.
* `extract()` returns a `Repository` object (instead of editing the extractor in place.
* `Repository` is a dataclass that stores the repository metadata and can be serialized to RDF.
* All serialization schemas are removed, as we only need `RepositorySchema`

Using a single serialization for all extractors required minor metadata alignments. I made the following choices:
* There can be multiple `authors` (git and github had a single author, whereas gitlab had multiple)
* Dropped `schema:isPartOf`, which was only used in GitLab
  + We also used it to denote affiliation to an organization, which is not its intended use

### CODE REVIEW

**Review by rmfranken — CHANGES_REQUESTED** [review]
tres beau.

File: gimie/sources/git.py [hunk 1]
```diff
@@ -59,42 +58,49 @@ class GitExtractor(Extractor):
...
-        g: Graph = Graph().parse(data=str(jd), format="json-ld")
-        g.bind("schema", SDO)
-        return g
+    def __del__(self):
+        """Cleanup the cloned repo."""
+        try:
+            if self.local_path is not None:
+                shutil.rmtree(self.local_path)
+        except AttributeError:
+            pass
```
[the remark this answers is not in the document]
Comment (cmdoret) [reply gimie/sources/git.py]: [9854534](https://github.com/SDSC-ORD/gimie/pull/96/commits/985453479c600ee8502f7f9015aa75dc3b4ed788) addresses the issue by:
* Only removing folder if it starts by $TMPDIR (os-dependent) AND it's been cloned by the extractor
* Removing a similar cleanup step previously handled by `gimie.Project` (no point in doing it twice)

[the remark this answers is not in the document]
Comment (rmfranken) [reply gimie/sources/git.py]: If I understand correctly, `self.local_path.startswith(tempfile.gettempdir())` is what checks the file is a tempfile. This is what makes the check OS independent right? The actual name given in Linux, Windows or MacOS will differ (OS dependent), but the filename can be compared to the result of `self.local_path.startswith(tempfile.gettempdir())`, assuming the user is not somehow using a foreign file structure in their OS .

Just checking that I understand this one - logically it makes sense. Very safe ;)

[the remark this answers is not in the document]
Comment (cmdoret) [reply gimie/sources/git.py]: The official doc of the function says:

> Python searches a standard list of directories to find one which the calling user can create files in. The list is:
> 
>     The directory named by the TMPDIR environment variable.
> 
>     The directory named by the TEMP environment variable.
> 
>     The directory named by the TMP environment variable.
> 
>     A platform-specific location:
> 
>         On Windows, the directories C:\TEMP, C:\TMP, \TEMP, and \TMP, in that order.
> 
>         On all other platforms, the directories /tmp, /var/tmp, and /usr/tmp, in that order.
> 
>     As a last resort, the current working directory.
>

**Review by rmfranken — APPROVED** [review]
So safe!

**Pull request merged**